src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+    private final int firstLeg;
+    private final int height;
+    private final int secondLeg;
+
+    public IsoscelesTrapezoid(int firstLeg, int secondLeg, int height, String color) {
+        this.firstLeg = firstLeg;
+        this.secondLeg = secondLeg;
+        this.height = height;
+        setColor(color);
+        obtainTheArea();
```

Comment:
Don't call 'obtainTheArea' in the constructor. It's better to call such methods when you actually need to get the area, not at the time of object creation. This can lead to better performance and avoid unnecessary calculations.